shared/src/main/scala/async/futures.scala

```diff
@@ -363,7 +362,7 @@ enum TaskSchedule:
 class Task[+T](val body: (Async, AsyncOperations) ?=> T):
 
   /** Start a future computed from the `body` of this task */
-  def run(using Async, AsyncOperations) = Future(body)
+  def run(using Async.Spawnable, AsyncOperations) = Future(body)
```

Comment:
This should use the same `Async`-capturing magic as `Future.apply`. I don't remember why we introduced `AsyncOperations`, though, and I guess it's not needed (anymore).

Reply:
> I don't remember why we introduced `AsyncOperations`

We need them for `sleep` needed to implement backoff/delay

Reply:
> Would it make more sense that `run` is
> 
> ```scala
> def run(using Async): T = // blocking 
> ```
> 
> instead? I feel like that would fall more in line with our APIs.

I totally agree. But maybe we could introduce both, similarly to `Thread`'s `run` and `start`. But note how `Task`s were introduced by Martin Odersky in the (still current) README:

> One criticism leveled against futures is that they "lack referential transparency". [...] Of course one can still achieve all that by working with functions producing futures instead of futures directly. [...] Tasks make the definition of such delayed futures a bit easier. (https://github.com/lampepfl/gears#tasks)